test/unit/core/http/retry_policy_test.rb

```diff
@@ -20,4 +20,14 @@
     retry_policy = Azure::Core::Http::RetryPolicy.new do |a,b| true end
     retry_policy.should_retry?(nil, nil).must_equal true
   end
+  
+  it 'uses retry policy as retry logic' do
+    retry_policy = Azure::Core::FixtureRetryPolicy.new
+    retry_policy.should_retry?(nil, {:error => 'Error: Retry'}).must_equal true
+  end
+
+  it 'uses retry policy as retry logic' do
```

Comment:
could you update the name of one of these tests so they describe the different cases (they are currently the same)?